CriticalMassTests/RatingHelperTests.swift

```diff
@@ -19,71 +19,96 @@ class MockRatingRequest: RatingRequest {
 
         XCTAssertEqual(MockRatingRequest.requestCounter, 0)
     }
 
     func testNoRequestAfterFiveLaunchesOnSameDay() {
+        ratingsStorage = RatingStorageMock()
+        ratingHelper = RatingHelper(
+            ratingStorage: ratingsStorage,
+            ratingRequest: MockRatingRequest.self
+        )
         execute(times: 5, ratingHelper.onLaunch())
 
         XCTAssertEqual(MockRatingRequest.requestCounter, 0)
     }
 
     func testRequestAfterFiveLaunchesOnNextDay() {
         // mock app used yesterday
-        userdefaults.lastDayUsed = .yesterday()
+        ratingsStorage = RatingStorageMock()
+        ratingsStorage.lastDayUsed = .yesterday()
+        ratingHelper = RatingHelper(
+            ratingStorage: ratingsStorage,
+            ratingRequest: MockRatingRequest.self
+        )
 
         execute(times: 5, ratingHelper.onLaunch())
         XCTAssertEqual(MockRatingRequest.requestCounter, 1)
     }
 
     func testNoRequestIfVersionAlreadyRated() {
         // mock app used yesterday
-        userdefaults.lastDayUsed = .yesterday()
-
-        // mock version already rated
-        userdefaults.lastRatedVersion = Bundle.main.versionNumber + Bundle.main.buildNumber
+        ratingsStorage = RatingStorageMock()
+        ratingsStorage.lastDayUsed = .yesterday()
+        ratingsStorage.lastRatedVersion = Bundle.main.versionNumber + Bundle.main.buildNumber
+        ratingHelper = RatingHelper(
+            ratingStorage: ratingsStorage,
+            ratingRequest: MockRatingRequest.self
+        )
 
         execute(times: 5, ratingHelper.onLaunch())
         XCTAssertEqual(MockRatingRequest.requestCounter, 0)
     }
 
     func testRatingRequestAfterInstallingNewVersion() {
+        ratingsStorage = RatingStorageMock()
         // mock app used yesterday
-        userdefaults.lastDayUsed = .yesterday()
-
+        ratingsStorage.lastDayUsed = .yesterday()
         // mock version already rated
-        userdefaults.lastRatedVersion = Bundle.main.versionNumber + Bundle.main.buildNumber
+        ratingsStorage.lastRatedVersion = Bundle.main.versionNumber + Bundle.main.buildNumber
+
+        ratingHelper = RatingHelper(
+            ratingStorage: ratingsStorage,
+            ratingRequest: MockRatingRequest.self
+        )
 
         execute(times: 5, ratingHelper.onLaunch())
         XCTAssertEqual(MockRatingRequest.requestCounter, 0)
 
         // mock app used yesterday
-        userdefaults.lastDayUsed = .yesterday()
+        ratingsStorage.lastDayUsed = .yesterday()
 
         // mock new version  installed
         ratingHelper = RatingHelper(
-            defaults: userdefaults,
+            ratingStorage: ratingsStorage,
             ratingRequest: MockRatingRequest.self,
             currentVersion: "MockVersion"
         )
         execute(times: 5, ratingHelper.onLaunch())
         XCTAssertEqual(MockRatingRequest.requestCounter, 1)
     }
 }
+
+struct RatingStorageMock: RatingStorage {
```

Comment:
Maybe it's a good idea to make all of the mock structs in tests private so it's only available in the same file?

Reply:
For some it is actually. Others are needed in more then one place